src/mesh/Router.cpp

```diff
@@ -248,15 +248,7 @@ ErrorCode Router::send(meshtastic_MeshPacket *p)
     // If the packet is not yet encrypted, do so now
     if (p->which_payload_variant == meshtastic_MeshPacket_decoded_tag) {
         ChannelIndex chIndex = p->channel; // keep as a local because we are about to change it
-
-        if (moduleConfig.mqtt.enabled) {
-
-            LOG_INFO("Should encrypt MQTT?: %d\n", moduleConfig.mqtt.encryption_enabled);
-
-            // the packet is currently in a decrypted state.  send it now if they want decrypted packets
-            if (mqtt && !moduleConfig.mqtt.encryption_enabled)
-                mqtt->onSend(*p, chIndex);
-        }
+        meshtastic_MeshPacket *p_decoded = p;
```

Comment:
Hmm, you're letting `p_decoded` point to the same memory location as `p` here. When `perhapsEncode()` modifies the memory pointed to by `p`, it also modifies where `p_decoded` points to. 

We'd need to make a copy, but then we also need to be careful that the memory gets released afterwards.

Reply:
thanks for catching that. fix incoming.